homeassistant/components/recorder.py

```diff
@@ -404,6 +410,29 @@ def save_migration(migration_id):
 
             save_migration(4)
 
+        if migration_id < 5:
+            # Add domain so that thermostat graphs look right
+            self.query("""
+                ALTER TABLE states
+                ADD COLUMN domain text
+            """)
+
+            # populate domain with defaults
+            self.query("""
+                UPDATE states
+                set domain=substr(entity_id, 0, instr(entity_id, '.'))
+            """)
+
+            # add indexes we are going to use a lot on selects
+            self.query(
```

Comment:
I use [SQLIte Manager](https://addons.mozilla.org/en-US/firefox/addon/sqlite-manager/) to explore different options and get query plans.